3rdparty/libcxxrt/CMakeLists.txt

```diff
@@ -1,16 +1,26 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
-# Compile CXXRT
+# Compile libcxx for Open Enclave. Some of these build rules emulate
+# those found in the original sources, but instead this builds an
+# OBJECT library in our CMake graph.
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-tautological-compare")
+add_library(libcxxrt OBJECT
+  libcxxrt/src/dynamic_cast.cc
+  libcxxrt/src/exception.cc
+  libcxxrt/src/guard.cc
+  libcxxrt/src/stdexcept.cc
+  libcxxrt/src/typeinfo.cc
+  libcxxrt/src/memory.cc
+  libcxxrt/src/auxhelper.cc
+  libcxxrt/src/libelftc_dem_gnu3.c)
 
-# build cxxrt sources passing additional options
-add_subdirectory(libcxxrt/src)
-target_compile_options(cxxrt-static PRIVATE
-    -fPIC
-    # cannot use add_link_libraries on out-of-dir target
-    -nostdinc $<$<COMPILE_LANGUAGE:CXX>:-nostdinc++>
-    -I${OE_INCDIR}/openenclave/libc
-    )
-add_dependencies(cxxrt-static oelibc_includes)
+set_property(SOURCE libcxxrt/src/exception.cc PROPERTY COMPILE_FLAGS -Wno-unused-variable)
+
+set_property(TARGET libcxxrt PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+target_compile_options(libcxxrt PRIVATE -Wno-tautological-compare)
+
+target_compile_definitions(libcxxrt PRIVATE -D_GNU_SOURCE)
+
+target_link_libraries(libcxxrt PUBLIC oelibc)
```

Comment:
We spoke offline, but I am documenting it here:

I've changed the dependencies of `libcxx`, `libcxxrt`, and `libunwind` to each depend on `oelibc` instead of `oelibc_includes` so that consumers of any of these (well, really consumers of `oelibcxx` which is comprised of these) will inherit the dependency on `oelibc`. It's true that as a static library, `oelibcxx` does not need nor use the symbols from `oelibc`, but each of these libraries cannot be used without `oelibc`, and depend on its `INTERFACE_INCLUDE_DIRECTORIES` and `INTERFACE_COMPILE_OPTIONS` etc. So we instead represent this in the graph via `target_link_libraries(libcxxrt PUBLIC oelibc)`: "when linking to libcxxrt you must also link to oelibc and use its interface options."